packages/@react-aria/menu/src/useMenuTrigger.ts

```diff
@@ -36,50 +39,81 @@ interface MenuTriggerAria {
  */
 export function useMenuTrigger(props: MenuTriggerAriaProps, state: MenuTriggerState, ref: RefObject<HTMLElement>): MenuTriggerAria {
   let {
-    type = 'menu' as MenuTriggerAriaProps['type']
+    type = 'menu' as MenuTriggerAriaProps['type'],
+    trigger
   } = props;
 
   let menuTriggerId = useId();
   let {triggerProps, overlayProps} = useOverlayTrigger({type}, state, ref);
+  
+  const handleArrowKeyBehaviour = (e) => {
+    switch (e.key) {
+      case 'ArrowDown':
+        e.preventDefault();
+        state.toggle('first');
+        break;
+      case 'ArrowUp':
+        e.preventDefault();
+        state.toggle('last');
+        break;
+    }
+  };
 
-  let onKeyDown = (e) => {
+  const handleLongPressAltBehaviour = (e) => {
+    if (e.altKey) {
+      handleArrowKeyBehaviour(e);
+    }
+  };
+  
+  const onKeyDown = (e) => {
     if ((typeof e.isDefaultPrevented === 'function' && e.isDefaultPrevented()) || e.defaultPrevented) {
       return;
     }
 
     if (ref && ref.current) {
-      switch (e.key) {
-        case 'ArrowDown':
-          e.preventDefault();
-          state.toggle('first');
-          break;
-        case 'ArrowUp':
-          e.preventDefault();
-          state.toggle('last');
-          break;
+      if (trigger === 'longPress') {
+        handleLongPressAltBehaviour(e);
+      } else  {
+        handleArrowKeyBehaviour(e);
       }
     }
   };
 
-  return {
-    menuTriggerProps: {
-      ...triggerProps,
-      id: menuTriggerId,
-      onPressStart(e) {
-        // For consistency with native, open the menu on mouse/key down, but touch up.
-        if (e.pointerType !== 'touch') {
-          // If opened with a keyboard or screen reader, auto focus the first item.
-          // Otherwise, the menu itself will be focused.
-          state.toggle(e.pointerType === 'keyboard' || e.pointerType === 'virtual' ? 'first' : null);
-        }
-      },
-      onPress(e) {
-        if (e.pointerType === 'touch') {
-          state.toggle();
-        }
-      },
-      onKeyDown
+  const longPressProps = useLongPress({
+    // Close on press start as menu can be in a open state after onLongPress.
+    onPressStart() {
+      state.close();
```

Comment:
Recent changes for fixing #925 updated the behaviour of useOverlay. Earlier, onBlurWithin used to get called which calls onClose but now the close is handled only by onPressStart from pressProps. (For trigger buttons)
To replicate a similar behavior, I added state.close.